component/src/main/java/org/wso2/carbon/identity/authenticator/FoursquareAuthenticator.java

```diff
@@ -30,322 +30,367 @@
+				// create OAuth client that uses custom http client under the hood
+				OAuthClient oAuthClient = new OAuthClient(new URLConnectionClient());
+				OAuthClientResponse oAuthResponse;
+				oAuthResponse = oAuthClient.accessToken(accessRequest);
+				String accessToken = oAuthResponse.getParam(FoursquareAuthenticatorConstants.ACCESS_TOKEN);
+				if (StringUtils.isNotEmpty(accessToken)) {
+					Map<ClaimMapping, String> claims = buildClaims(oAuthResponse, authenticatorProperties);
+					if (claims != null && !claims.isEmpty()) {
+						//Find the subject from the IDP claim mapping, subject Claim URI.
+						String subjectFromClaims = FrameworkUtils
+								.getFederatedSubjectFromClaims(context.getExternalIdP().getIdentityProvider(), claims);
+						associateSubjectFromClaims(context, subjectFromClaims, claims);
+					} else {
+						throw new AuthenticationFailedException(
+								"Claims for the user not found for access Token : " + accessToken);
+					}
+				} else {
+					throw new AuthenticationFailedException("Could not receive a valid access token from Foursquare");
+				}
+			} catch (OAuthSystemException e) {
+				throw new AuthenticationFailedException("Exception while building access token request", e);
+			} catch (ApplicationAuthenticatorException e) {
+				throw new AuthenticationFailedException("Exception while building the claim mapping", e);
+			}
+		} catch (OAuthProblemException e) {
+			throw new AuthenticationFailedException("Exception while getting the access token form the response", e);
+		}
+	}
+
+	/**
+	 * This method is to get the Foursquare user details.
+	 *
+	 * @param url         user info endpoint.
+	 * @param accessToken access token.
+	 * @return user info
+	 * @throws ApplicationAuthenticatorException
+	 */
+	private JSONObject fetchUserInfo(String url, String accessToken) throws ApplicationAuthenticatorException {
+		if (log.isDebugEnabled()) {
+			log.debug("Sending the request for getting the user info");
+		}
+		StringBuilder jsonResponseCollector = new StringBuilder();
+		BufferedReader bufferedReader = null;
+		HttpURLConnection httpConnection = null;
+		JSONObject jsonObj = null;
+		try {
+			URL obj = new URL(url + "&" + FoursquareAuthenticatorConstants.FOURSQUARE_OAUTH2_ACCESS_TOKEN_PARAMETER +
+			                  "=" + accessToken);
+			URLConnection connection = obj.openConnection();
+			// Cast to a HttpURLConnection
+			if (connection instanceof HttpURLConnection) {
+				httpConnection = (HttpURLConnection) connection;
+				httpConnection.setConnectTimeout(FoursquareAuthenticatorConstants.CONNECTION_TIMEOUT_VALUE);
+				httpConnection.setReadTimeout(FoursquareAuthenticatorConstants.READ_TIMEOUT_VALUE);
+				httpConnection.setRequestMethod(FoursquareAuthenticatorConstants.HTTP_GET_METHOD);
+				bufferedReader = new BufferedReader(new InputStreamReader(httpConnection.getInputStream()));
+			} else {
+				throw new ApplicationAuthenticatorException("Exception while casting the HttpURLConnection");
```

Comment:
Fixed in https://github.com/wso2-extensions/identity-outbound-auth-foursquare/pull/2/commits/58520246ef2e797650ce96df5d6f5fbde73cfa2d